src/libexpr/symbol-table.hh

```diff
@@ -140,6 +140,6 @@ struct std::hash<nix::Symbol>
 {
     std::size_t operator()(const nix::Symbol & s) const noexcept
     {
-        return std::hash<decltype(s.id)>{}(s.id);
+        return s.id;
```

Comment:
Symbol is a counter, meaning its high bits are all zero. When Symbol::id is directly used as a hash value, this could lead to poor distribution in hash tables, causing performance degradation due to increased collisions.

Even if your testing does not reveal a performance problem now, I'd prefer to be cautious and keep a cheap hash function, so that we don't have to assess this for libcxx and msvc++.